src/modules/asab-config/ConfigEditor.js

```diff
@@ -0,0 +1,222 @@
+	const Axios = props.app.axiosCreate(url);
+
+	useEffect(() => {
+		load();
+	},[]);
+
+
+	const load = async () => {
+		if (typeId) {
+			await Axios.get("/type/" + typeId).then(response => {
+				setType(response.data);
+			})
+			.catch(error => {
+				console.log(error); // log the error to the browser's console
+				App.addAlert("warning", t(`Unable to get ${typeId} data: `, { error: error.toString() }));
+			});
+		}
+	}
+
+	useEffect(() => {
+
+		async function fetchValues() {
+			// let config_response = await axios.get("/config/new_type/my_config")
+			// let values = config_response.data
+
+			// Mocked
+			let values = {
+				'general': {
+					'uid': "123",
+				}
+			};
+
+			for (var section in values) {
+				for (var key in values[section]) {
+					setValue('['+section + "] " + key, values[section][key], { shouldValidate: false })
+				}
+			}
+		}
+		fetchValues();
+
+	},[type]);
+
+
+	return (
+		<Container fluid className="animated fadeIn flex">
+			<Row>
+				<Col md={{ size: 6, offset: 3 }}>
+					<Form onSubmit={handleSubmit(onSubmit)}>
+						
+						<Card style={{marginBottom: "0.25em"}}>
+							<div style={{margin: "0.5em"}}>
+								<Button
+									color="primary"
+									type="submit"
+								>
+									Save
+								</Button>
+								<div className="float-right">
+									<span className="cil-briefcase pr-3" />
+									{typeId}
+								</div>
+							</div>
+						</Card>
+
+						{type && type.properties && Object.keys(type.properties).map((section_name, idx) =>
+							<ConfigSection
+								key={idx}
+								section={type.properties[section_name]}
+								sectionname={section_name}
+								register={register}
+							/>
+						)}
+
+						{/* TODO: List all remaining sections in "values" as a  ConfigAdHocSection */}
+
+					</Form>
+				</Col>
+			</Row>
+		</Container>
+	);
+}
+
+
+function ConfigSection(props) {
+	const [isOpen, setIsOpen] = useState(false);
+	const toggle = () => setIsOpen(!isOpen);
+
+	return (
+		<Card style={{marginBottom: "0.25em"}}>
+			<CardHeader tag="h5" onClick={toggle}>
+				{props.section['title']}
+			</CardHeader>
+			<Collapse isOpen={isOpen}>
+				<CardBody>
+					{Object.keys(props.section.properties).map((item_name, idx) =>
+						// TODO: Decide that type of config item to render
+						<ConfigItem
+							key={idx}
+							item={props.section.properties[item_name]}
+							itemname={item_name}
+							sectionname={props.sectionname}
+							register={props.register}
+						/>
+					)}
+
+					{/* TODO: List all remaining key/values from a config as simple Config Item  */}
+
+				</CardBody>
+			</Collapse>
+		</Card>
+	);
+}
+
+// TODO: Different types of ConfigItem to cover formats such as "number", "boolean", checkbox, radiobox
+function ConfigItem(props) {
+	let myid = '['+props.sectionname + "] " + props.itemname;
+	return (
+		<FormGroup>
+			<Label for={myid}>
+				{props.item['title']}
+			</Label>
+			<ConfigItemFormats
+				format={props.item['type']}
```

Comment:
@ateska Hope this is not much of stupid question, but will be input formats taken from `type` property or will be there any format property in the schema to distinguish, which format to take into consideration? I updated the component `ConfigItem` with `ConfigItemFormats` and this require to pass to props a format type, to select the proper one. I set it to the `type` property for now, but it can be easily changed. So my question is, if I should substract it from `type` property, or is it defined somewhere else in the schema?

Reply:
Well, I rather expect various specializations of `ConfigItem` .... eg `<EmailConfigItem>` or `<CheckboxConfigItem>`.

The type should be based on "some" property in the schema ... and this shoud be so far relatively easy mapping (e.g. by default, it is always `ConfigItem`, if specific format is requested, then you use more specific ConfigItem`.